src/main/java/org/meteordev/starscript/utils/Stack.java

```diff
@@ -1,37 +1,54 @@
 package org.meteordev.starscript.utils;
 
+import java.util.Arrays;
+
 public class Stack<T> {
-    @SuppressWarnings("unchecked")
-    private T[] items = (T[]) new Object[8];
+    private Object[] items = new Object[8];
```

Comment:
By making the array an `Object` array instead of a `T` array, you end up casting on every operation instead of once on array creation

Reply:
To be fair, T[] does compile to Object[] and casts on every operation anyway